runtime-sdk/modules/evm/src/evm_backend.rs

```diff
@@ -166,82 +148,45 @@ impl<'c, C: oasis_runtime_sdk::Context> ApplyBackend for Backend<'c, C> {
                     reset_storage,
                 } => {
                     let addr: H160 = address.into();
-                    let is_empty = {
+                    // Derive SDK account address from the Ethereum address.
+                    let address = Cfg::map_address(address);
+
+                    // Update account balance and nonce.
+                    let mut state = self.ctx.get_mut().runtime_state();
+                    let amount =
+                        token::BaseUnits::new(basic.balance.as_u128(), Cfg::TOKEN_DENOMINATION);
+                    Cfg::Accounts::set_balance(&mut state, address, &amount);
+                    Cfg::Accounts::set_nonce(state, address, basic.nonce.as_u64());
+
+                    // Handle code updates.
+                    if let Some(code) = code {
                         let state = self.ctx.get_mut().runtime_state();
-                        let a_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                        let mut accounts = storage::TypedStore::new(storage::PrefixStore::new(
-                            a_store,
-                            &state::ACCOUNTS,
-                        ));
-                        let mut account: Account = accounts.get(&addr).unwrap_or_default();
-
-                        account.balance = basic.balance.into();
-                        account.nonce = basic.nonce.into();
-
-                        accounts.insert(&addr, account.clone());
-
-                        if let Some(code) = code {
-                            let state = self.ctx.get_mut().runtime_state();
-                            let c_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let mut codes = storage::TypedStore::new(storage::PrefixStore::new(
-                                c_store,
-                                &state::CODES,
-                            ));
-                            codes.insert(&addr, code);
-                        }
+                        let mut store = state::codes(state);
+                        store.insert(&addr, code);
+                    }
 
-                        if reset_storage {
-                            let state = self.ctx.get_mut().runtime_state();
-                            let s_store = storage::PrefixStore::new(state, &crate::MODULE_NAME);
-                            let mut storages = storage::PrefixStore::new(s_store, &state::STORAGES);
-                            storages.remove(addr.as_bytes());
-                        }
+                    // Handle storage reset.
+                    if reset_storage {
+                        // NOTE: Storage cannot be efficiently reset as this would require iterating
+                        //       over all of the storage keys. We could add this if remove_prefix
+                        //       existed.
```

Comment:
so we do nothing? is it correct for us to do that?

what did the removed code do?